pkg/registry/common/memory/ns_server.go

```diff
@@ -103,7 +106,7 @@ func (s *memoryNSServer) Find(query *registry.NetworkServiceQuery, server regist
 	var err error
 	for ; err == nil; err = s.receiveEvent(query, server, eventCh) {
 	}
-	if err != io.EOF {
+	if errors.Cause(err).Error() != io.EOF.Error() {
```

Comment:
Please apply this for other places.

```suggestion
	if errors.Is(err, io.EOF) {
```